src/containers/task-group.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Task from '../components/task';
+
+class TaskGroup extends Component {
+
+  processTaskList(list, givenMonth, givenType) {
+    let resultsArray = [];
+    const loop = list.map( (item) => {
+      if(item.month === givenMonth && item.type === givenType) {
+        resultsArray.push(item);
+      };
+    });
+    if(resultsArray.length) {
+      return true;
+    }
+  }
+
+  taskTypeIsUsed(list, givenMonth, givenType) {
+    let taskIsUsed = false;
+
+    // verification is an array containing lots of
+    // 'undefined' + occasionally true boolean
+    const verification = list.map( (item) => {
+      if(this.processTaskList(item.tasks, givenMonth, givenType) !== undefined) {
+        taskIsUsed = true;
+      }
+    });
+
+    return taskIsUsed;
+  };
+
+  taskHasItems(list, givenMonth, givenType) {
+    return this.processTaskList(list, givenMonth, givenType);
+  };
+
+  renderTask(taskType) {
+    const taskClass = taskType.toLowerCase().replace(" ", "-");
+
+    if(this.taskTypeIsUsed(this.props.data, this.props.thisMonth, taskType)) {
```

Comment:
I mean what you wrote, but without putting it in an else block; since the if branch returns, you can just write out the rest as the rest of the method.